src/System.Private.CoreLib/src/System/Array.Portable.cs

```diff
@@ -0,0 +1,1659 @@
+        }
+
+        public static void Sort(Array keys, Array items, int index, int length)
+        {
+            Sort(keys, items, index, length, null);
+        }
+
+        public static void Sort(Array keys, Array items, int index, int length, IComparer comparer)
+        {
+            if (keys == null)
+                throw new ArgumentNullException(nameof(keys));
+            if (keys.Rank != 1 || (items != null && items.Rank != 1))
+                throw new RankException(SR.Rank_MultiDimNotSupported);
+            int keysLowerBound = keys.GetLowerBound(0);
+            if (items != null && keysLowerBound != items.GetLowerBound(0))
+                throw new ArgumentException(SR.Arg_LowerBoundsMustMatch);
+            if (index < keysLowerBound || length < 0)
+                throw new ArgumentOutOfRangeException((length < 0 ? nameof(length) : nameof(index)), SR.ArgumentOutOfRange_NeedNonNegNum);
+            if (keys.Length - (index - keysLowerBound) < length || (items != null && (index - keysLowerBound) > items.Length - length))
+                throw new ArgumentException(SR.Argument_InvalidOffLen);
+
+            if (length > 1)
+            {
+#if CORERT
+                IComparer<Object> comparerT = new ComparerAsComparerT(comparer);
+                Object[] objKeys = keys as Object[];
+                Object[] objItems = items as Object[];
+
+                // Unfortunately, on Project N, we don't have the ability to specialize ArraySortHelper<> on demand
+                // for value types. Rather than incur a boxing cost on every compare and every swap (and maintain a separate introsort algorithm
+                // just for this), box them all, sort them as an Object[] array and unbox them back.
+
+                // Check if either of the arrays need to be copied.
+                if (objKeys == null)
+                {
+                    objKeys = new Object[index + length];
+                    Array.CopyImplValueTypeArrayToReferenceArray(keys, index, objKeys, index, length, reliable: false);
+                }
+                if (objItems == null && items != null)
+                {
+                    objItems = new Object[index + length];
+                    Array.CopyImplValueTypeArrayToReferenceArray(items, index, objItems, index, length, reliable: false);
+                }
+
+                Sort<Object, Object>(objKeys, objItems, index, length, comparerT);
+
+                // If either array was copied, copy it back into the original
+                if (objKeys != keys)
+                {
+                    Array.CopyImplReferenceArrayToValueTypeArray(objKeys, index, keys, index, length, reliable: false);
+                }
+                if (objItems != items)
+                {
+                    Array.CopyImplReferenceArrayToValueTypeArray(objItems, index, items, index, length, reliable: false);
+                }
+#else
+                Object[] objKeys = keys as Object[];
```

Comment:
This is going to break us on the TFS side because Project N (the AOT compiler for Universal Windows Apps) doesn't define CORERT.